### ISSUE TITLE
Questions about Mimi: loss balancing and bfloat16/mixed precision training.

### ISSUE THREAD

**Issue by SarthakYadav** [kyutai-labs/moshi#178]
### Due diligence

- [X] I have done my due diligence in trying to find the answer myself.

### Topic

The paper

### Question

Thanks for the great work. I'm trying to reproduce Mimi and had the following questions:

1. Does Mimi use a loss balancer such as that used in Encodec for training? The paper points to the default Encodec configuration in AudioCraft which uses loss balancing, so I was wondering if that's the case for Mimi as well.
2. Was Mimi trained in bfloat16? Or did the actual training happen in full precision and the weights were exported in bfloat16?

Thanks!

**Comment by akshatvishu** [comment]
@SarthakYadav it seems the training code for Mimi has not yet been released, but they plan to do so in the near future, as mentioned in their FAQ section. However, here's what I could gather from the current resources:  
In-their [readme.md](<https://github.com/kyutai-labs/moshi/blob/main/README.md>) page, they state :

> Finally, and similarly to EBEN, Mimi uses only an adversarial training loss, along with feature matching, showing strong improvements in terms of subjective quality despite its low bitrate.

so, i personally don't think they've used a loss balancer.

Asfor Q2,  I think only the official team could  answer as  they've not released the training code yet!

**Comment by LaurentMazare** [comment]
> 2\. Was Mimi trained in bfloat16? Or did the actual training happen in full precision and the weights were exported in bfloat16?

Which weights are you referring to? Looking at the `model.safetensors` file on our [huggingface repo](https://huggingface.co/kyutai/mimi/tree/main), the weights should actually be in fp32 rather than bf16 (and this should be the case too in our other repos).

**Comment by SarthakYadav** [comment]
@LaurentMazare Thanks, tokenizer weights are indeed fp32, it's only the moshi weights that are bf16.
